extensions/datasource/deployment/src/main/java/io/quarkus/datasource/deployment/devservices/DevServicesDatasourceProcessor.java

```diff
@@ -77,13 +79,17 @@ DevServicesDatasourceResultBuildItem launchDatabases(
         //if not and the DB's have already started we just return
         if (databases != null) {
             boolean restartRequired = false;
-            if (!restartRequired) {
-                for (Map.Entry<String, String> entry : cachedProperties.entrySet()) {
-                    if (!Objects.equals(entry.getValue(),
-                            trim(ConfigProvider.getConfig().getOptionalValue(entry.getKey(), String.class).orElse(null)))) {
-                        restartRequired = true;
-                        break;
-                    }
+            for (Map.Entry<String, String> entry : cachedProperties.entrySet()) {
+                String value = ConfigProvider.getConfig()
+                        .getOptionalValue(entry.getKey(), String.class)
+                        .orElse(null);
+                if (value == null) {
+                    value = getValueWithOrWithoutQuotes(entry.getKey(),
```

Comment:
I'm not blocking the merge; if it works let's merge.

But! Ideally we would instead:

1. (maybe) make sure to always normalize keys (use quotes) before putting them in `cachedProperties`
2. make sure that `getOptionalValue` does not differentiate between quoted and non-quoted syntax

@radcortez any opinion/insight about item 2? It would have made my fix in #26815 unnecessary.

Reply:
When calling a `get` in the config system with the full property name, we require an exact match because we cannot tell if the quotes should be omitted or not. In most cases, we can't even control that because the source provides the value (and all sources do exact matches).

Alternatively, we could look the configuration as a `Map`, and in that case, the config system is smart enough to remove the quotes from the path key segments. You could do a `SmallRyeConfig#getOptionalValues("quarkus.datasource", String.class, String.class)`.

Reply:
Thanks for the feedback i pushed a new commit using `SmallRyeConfig#getOptionalValues` to create the cache which simplies this a lot.